app/src/main/java/com/asma/tasky/feature_management/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,20 @@
+package com.asma.tasky.feature_management.data.repository
+
+import com.asma.tasky.feature_management.data.data_source.TaskyDao
+import com.asma.tasky.feature_management.data.mapper.toAgendaTask
+import com.asma.tasky.feature_management.domain.AgendaItem
+import com.asma.tasky.feature_management.domain.agenda.repository.AgendaRepository
+import java.time.LocalDate
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class AgendaRepositoryImpl(private val dao: TaskyDao) : AgendaRepository {
+
+    override fun getTasksOfTheDay(day: LocalDate): Flow<List<AgendaItem.Task>> {
+        return dao.getTasksOfTheDay().map { tasks ->
+            tasks.map {
+                it.toAgendaTask()
+            }
+        }
+    }
```

Comment:
Not sure if it already exists in your app, but you can have one function that returns all agenda items for a day by using `combine`.
```
combine(dao.getTasksOfTheDay(), dao.getEventsOfTheDay(), dao.getRemindersOfTheDay()) { tasks, events, reminders -> {
    // Executed when either of these emits something
    (tasks + events + reminders).sortedBy { it.startDate }
}
```